beacon_node/beacon_chain/src/builder.rs

```diff
@@ -1037,7 +1037,10 @@ where
         );
 
         // Check for states to reconstruct (in the background).
-        if beacon_chain.config.reconstruct_historic_states {
+        if beacon_chain.config.reconstruct_historic_states
+            && beacon_chain.genesis_backfill_slot == 0
```

Comment:
`genesis_backfill_slot` is always set to 0 if `reconstruct_historic_states` is true right? Is the extra condition for `genesis_backfill_slot == 0 ` necessary?

Reply:
good point! I'll remove it